### DISCUSSION TITLE
Chord Graphs

### DISCUSSION THREAD

**Discussion by arthursoares** [no-chris/chord-mark#633] [Ideas]
First of all, congratulations for the awesome tool. I've been meaning for a replacement for ChordPro and your approach hits very close to home. I really enjoy the work you did with chord durations. I've tried hacking my way with ChordPro and it never felt really there.

One thing that I'd love to see or hear if it is in your plans is to be able to show Chord Charts/Graphs and customize chords.

I'm mostly working with Bossa Nova transcriptions where the Chords have very specific voicings. I would love to be able to export something like this:

<img width="646" alt="Capture 2023-04-17 00 21 40" src="https://user-images.githubusercontent.com/533339/232345919-0657ffaf-049b-43a6-9fd6-833dbd281fed.png">

**Comment by no-chris** [comment]
Hi Arthur! 

Thanks for dropping by. I'm glad to see more people interested in solving the same issues that I had when building ChordMark initially 😄 

Yes, showing chord fingering is definitely a must have. In that sense, I do "have plans" for it, but nothing more specific at that stage. ChordMark being a pet project, it only evolves when I get the chance to dedicate some time to it, which is not very much lately, having a toddler to take care of 😅 

Still, defining the syntax of new features is something that can already be done. Having user input is very valuable and leads to better solutions, a bit like we did [here](https://github.com/no-chris/chord-mark/discussions/500) for defining the sub-beat chord duration syntax.

Do you have any ideas/preferences on what the syntax should be?

For me the requirements of displaying chord graphs should be the following:

- support for multiple instruments (guitar, uke, piano...)
- support for multiple tunings of a given instrument
- support for multiple voicing of a given chord in the same song
- support for default voicings (from a chords fingering bank) or custom ones

Do you agree and/or see anything else?

**Comment by arthursoares** [comment]
Congratulation on the toddler! I can only imagine how it is to juggle a baby and a pet project :)

In terms of the syntax, I don't have specifications. I agree with the essential requirements that you've listed there. I would add [Chord!](https://getchord.com/) as a reference for managing a Chord voicings/dictionary.

A requirement that I would add is not only the voicings but different fingering of the same voicing. João Gilberto often played the same chord voicing in different regions of the guitar scale, creating different tone registers.

![CleanShot 2023-04-18 at 23 58 26@2x](https://user-images.githubusercontent.com/533339/232914472-f1dfa282-1950-4758-9a8f-d5aa44f74718.png)

Above is a simple example with the `A#m7` chord. 

---

Similarly to [SongBook](https://apps.apple.com/de/app/songbook/id534978338?mt=12), I'd support for picking a chord voicing from a dictionary or adding a custom one.
